File: Data Structures & Algorithms/is-anagram/submission-4.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size()!=t.size())
        {
            return false;
        }
        unordered_map<char,int>data;
        for(char ch:s)
        {
            data[ch]+=1;
        }
        for(char ch:t)
        {
            if(data.find(ch)==data.end() || data[ch]==0)
            {
                return false;
            }
            data[ch]-=1;
        }
        return true;

    }
};

File: Data Structures & Algorithms/anagram-groups/submission-4.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        data={}

        for s in strs:
            sort="".join(sorted(s))
            print(sort)
            if sort not in data:
                data[sort]=[]
            data[sort].append(s)

        return list(data.values())